src/main/java/net/countercraft/movecraft/combat/features/tracking/events/CraftSunkByEvent.java

```diff
@@ -56,7 +59,16 @@ public String causesToString() {
         StringBuilder stringBuilder = new StringBuilder();
         stringBuilder.append(((PlayerCraft) this.craft).getPilot().getDisplayName());
         stringBuilder.append(" ").append(I18nSupport.getInternationalisedString("Killfeed - Sunk By")).append(" ");
+
+        Player cause = (Player) latestDamage.getCause();
+        //The contains key method is to check for null for people who haven't run the command yet.
+        if (DamageTracking.getKillCredMap().containsKey(cause) && DamageTracking.getKillCredMap().get(cause) == true) {
+            //Scrambles the cause's name.
+            stringBuilder.append(ChatColor.MAGIC + latestDamage.getCause().getName());
+            return stringBuilder.toString();
+        }
         stringBuilder.append(NameUtils.offlineToName(latestDamage.getCause()));
```

Comment:
With the new changes to the DamageTracking, we could change this to something similar to the following:
```java
if (damageTracking.getKillCred(cause))
    stringBuilder.append(ChatColor.MAGIC);

stringBuilder.append(cause.getName());
stringBuilder.append(ChatColor.RESET);

Reply:
Just to be sure, I could do this for the if statement to also check for the bypass permission right? For some reason in my testing even if both are set to true, they don't do the append. 
![image](https://user-images.githubusercontent.com/68086172/164223379-ad7505b0-507d-484c-98b9-44e4dd14face.png)